src/VisualStudio/Core/Impl/CodeModel/FileCodeModel_Events.cs

```diff
@@ -20,23 +20,22 @@ public sealed partial class FileCodeModel
         private const int ElementDeletedDispId = 3;
         private const int ElementDeletedDispId2 = 4;
 
-        public bool FireEvents()
+        public void FireEvents()
         {
-            _codeElementTable.CleanUpDeadObjects();
+            _ = _codeElementTable.CleanUpDeadObjectsAsync().ReportNonFatalErrorAsync();
```

Comment:
can you just doc the fire-forget nature here?  (i don't really have a good sense of what to write.  but mostly because i don't understand this code anyways :)

Reply:
is there any concern about somehow ending up in a state where we have N of tehse queuing up somehow?

Reply:
> is there any concern about somehow ending up in a state where we have N of tehse queuing up somehow?

Since this call occurs on the UI thread, it should fast path through the `_needsCleanUp` check if several events come in together. In theory they could get queued up if `Add` is repeatedly called `CleanUpThreshold` (or more) times during a single cleanup operation, which seems unlikely.